src/test/java/org/jboss/logmanager/handlers/SocketHandlerTests.java

```diff
@@ -115,71 +111,81 @@ public void testTcpPortChange() throws Exception {
     @Test
     public void testProtocolChange() throws Exception {
         Assume.assumeFalse(JDK_8219991_ERROR_MESSAGE, JDK_8219991);
-        try (SocketHandler handler = createHandler(Protocol.TCP)) {
-            try (SimpleServer server = SimpleServer.createTcpServer(port)) {
-                final ExtLogRecord record = createLogRecord("Test TCP handler");
-                handler.doPublish(record);
-                final String msg = server.timeoutPoll();
-                Assert.assertNotNull(msg);
-                Assert.assertEquals("Test TCP handler", msg);
-            }
+        // keep handler for whole test
+        SocketHandler handler;
+        try (SimpleServer server = SimpleServer.createTcpServer()) {
+            handler = createHandler(Protocol.TCP, server.getPort());
+            final ExtLogRecord record = createLogRecord("Test TCP handler");
+            handler.doPublish(record);
+            final String msg = server.timeoutPoll();
+            Assert.assertNotNull(msg);
+            Assert.assertEquals("Test TCP handler", msg);
+        }
+        // wait until the OS really release used port. https://issues.redhat.com/browse/LOGMGR-314
+        Thread.sleep(50);
```

Comment:
Is this needed? It feels like a guess at best TBH and feels fragile.

Reply:
I think too this should not be here. But I didn't find way how to safely verify a port is free. All ways which i found are similar to: 

```
try (Socket ignored = new Socket("localhost", port)) {
        return false;
    } catch (ConnectException e) {
        return true;
    }
```
And we don't have guarantees after end of try with resources or after close method is a port really free. Sometimes isn't - especially when the CPU is utilized to 100% and these tests are started.